tests/paramgrill.c

```diff
@@ -340,76 +808,270 @@ static int BMK_seed(winnerInfo_t* winners, const ZSTD_compressionParameters para
+            case slog_ind: ptr->searchLog    = ind % SLOG_RANGE + ZSTD_SEARCHLOG_MIN;                             ind /= SLOG_RANGE; break;
+            case slen_ind: ptr->searchLength = ind % SLEN_RANGE + ZSTD_SEARCHLENGTH_MIN;                          ind /= SLEN_RANGE; break;
+            case tlen_ind: ptr->targetLength = tlen_table[(ind % TLEN_RANGE)];                                    ind /= TLEN_RANGE; break;
+        }
+    }
+}
+
+/* Initialize memotable, immediately mark redundant / obviously infeasible params as such */
+static void createMemoTable(U8* memoTable, ZSTD_compressionParameters paramConstraints, const constraint_t target, const varInds_t* varyParams, const int varyLen, const size_t srcSize) {
+    size_t i;
+    size_t arrayLen = memoTableLen(varyParams, varyLen);
+    int cwFixed = !paramConstraints.chainLog || !paramConstraints.windowLog;
+    int scFixed = !paramConstraints.searchLog || !paramConstraints.chainLog;
+    int whFixed = !paramConstraints.windowLog || !paramConstraints.hashLog;
+    int wFixed = !paramConstraints.windowLog;
+    int j = 0;
+    memset(memoTable, 0, arrayLen);
+    cParamZeroMin(&paramConstraints);
+
+    for(i = 0; i < arrayLen; i++) {
+        memoTableIndInv(&paramConstraints, varyParams, varyLen, i);
+        if(ZSTD_estimateCStreamSize_usingCParams(paramConstraints) > (size_t)target.cMem) {
+            memoTable[i] = 255;
+            j++;
+        }
+        if(wFixed && (1ULL << paramConstraints.windowLog) > srcSize) {
+            memoTable[i] = 255;
+        }
+        /* nil out parameter sets equivalent to others. */
+        if(cwFixed/* at most least 1 param fixed. */) {
+            if(paramConstraints.strategy == ZSTD_btlazy2 || paramConstraints.strategy == ZSTD_btopt || paramConstraints.strategy == ZSTD_btultra) {
+                if(paramConstraints.chainLog > paramConstraints.windowLog + 1) {
+                    if(memoTable[i] != 255) { j++; }
+                    memoTable[i] = 255;
+                }
+            } else {
+                if(paramConstraints.chainLog > paramConstraints.windowLog) {
+                    if(memoTable[i] != 255) { j++; }
+                    memoTable[i] = 255;
+                }
+            }
+        }
+
+        if(scFixed) {
+            if(paramConstraints.searchLog > paramConstraints.chainLog) {
+                if(memoTable[i] != 255) { j++; }
+                memoTable[i] = 255;
+            }
+        }
+
+        if(whFixed) {
+            if(paramConstraints.hashLog > paramConstraints.windowLog + 1) {
+                if(memoTable[i] != 255) { j++; }
+                memoTable[i] = 255;
+            }
+        }
+    }
+    DEBUGOUTPUT("%d / %d Invalid\n", j, (int)i);
+    if((int)i == j) {
+        DEBUGOUTPUT("!!!Strategy %d totally infeasible\n", (int)paramConstraints.strategy)
+    }
+}
+
+/* frees all allocated memotables */
+static void memoTableFreeAll(U8** mtAll) {
```

Comment:
If this function is the inverse of `createMemoTableArray()`,
it should be called `freeTableArray()`.